src/Compilers/Core/Portable/Collections/DictionaryExtensions.cs

```diff
@@ -106,5 +106,56 @@ public static ImmutableSegmentedDictionary<K, ImmutableArray<V>> ToImmutableSegm
             dictionary.Free();
             return result.ToImmutable();
         }
+
+        /// <summary>
+        /// Removes entries from a dictionary based on a specified condition. The condition is defined by a function that
+        /// evaluates each key-value pair.
+        /// </summary>
+        public static void RemoveAll<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, Func<TKey, TValue, bool> predicate)
+            where TKey : notnull
+            => RemoveAll(dictionary, static (key, value, predicate) => predicate(key, value), predicate);
```

Comment:
i always love this pattern of passing the predicate along as the arg into another predicate :D